### PR TITLE
Concretization cleanup

### PR DESCRIPTION
**Pull request by jagot**
Before starting to look at #182, I refactored the concretization, since so much was in common between the different array types.

Impacts #173.

### CODE REVIEW

File: src/derivative_operators/concretization.jl [hunk 1]
```diff
@@ -1,123 +1,60 @@
...
+LinearAlgebra.Array(A::DerivativeOperator{T}, N::Int=A.len) where T =
+    copyto!(zeros(T, N, N+2), A, N)
 
-    if use_winding(A)
-        stencil_pivot = 1 + A.stencil_length%2
-    else
-        stencil_pivot = div(stencil_length,2)
-    end
+SparseArrays.SparseMatrixCSC(A::DerivativeOperator{T}, N::Int=A.len) where T =
+    copyto!(spzeros(T, N, N+2), A, N)
```
Comment (ChrisRackauckas) [inline src/derivative_operators/concretization.jl]: This would be really slow?

Comment (jagot) [reply src/derivative_operators/concretization.jl]: Why would it? I overloaded `Base.copyto!` at the top of the file, the code of which is exactly the same as was in all three versions for different target matrices.

Comment (ChrisRackauckas) [reply src/derivative_operators/concretization.jl]: Adding to a sparse matrix using indexing is really slow: the indices should all be declared up front.

Comment (ChrisRackauckas) [reply src/derivative_operators/concretization.jl]: oh I see, ehh, it's not any worse than before, but it's still bad. If no one wants to actually do the `(I,J,V)`, then it's best to probably just build the banded matrix and then convert.

Comment (jagot) [reply src/derivative_operators/concretization.jl]: Sure :) I was anyway interested in the banded versions, but could not keep my hands off refactoring this.

Comment (ChrisRackauckas) [reply src/derivative_operators/concretization.jl]: Yeah, 99% of the time people should just call `sparse` and get the banded version.

**Pull request merged**